Sources/Diff/CollectionViewChanges.swift

```diff
@@ -0,0 +1,225 @@
+//
+//  Copyright © 2019 Rosberry. All rights reserved.
+//
+
+public final class CollectionViewDeleteInsert<T>: CustomStringConvertible {
```

Comment:
Personally, It'd add separate types for deletions and insertions. They may just be subclasses of `CollectionViewDeleteInsert` which will be their parent.
Like:
```
public final class CollectionViewInsert: CollectionViewDeleteInsertBase {}
public final class CollectionViewDelete: CollectionViewDeleteInsertBase {}
```

It'll provide some compile-time checking that deletion is not accidentally passed as insert.

Reply:
👌